### PR TITLE
Add mkdocstrings

### LINKED ISSUE #7100: Add PyDocs to the documentation
### Feature Request / Improvement

In PyIceberg we have quite extensive docstrings, but we don't expose them in forms of documentation. It would be awesome to have something like [mkdocstrings](https://github.com/mkdocstrings/mkdocstrings) (because we use `mkdocs` for generating the docs) to generate documentation from these docstrings.

### Query engine

None

### PR DESCRIPTION
**Pull request by LuigiCerone**
Closes #7100.

### CODE REVIEW

File: python/mkdocs/requirements.txt [hunk 1]
```diff
@@ -17,3 +17,6 @@
 
 mkdocs==1.4.2
 jinja2==3.1.2
+mkdocstrings==0.20.0
+mkdocstrings-python==0.8.3
+mkdocs-literate-nav==0.6.0
```
Comment (Fokko) [inline python/mkdocs/requirements.txt:22]: I had to install `mkdocs-gen-files==0.4.0` as well.

**Comment by Fokko** [comment]
@LuigiCerone what do you think? It feels a bit whacky on my end. Some pages are empty (for example Manifest), and all the menus are expanded which doesn't give a lot of useful information (I don't really care about the zstd decompression of Avro, and the page is also empty).

**Comment by LuigiCerone** [comment]
Hello @Fokko , I agree with you.

The reason for the blank pages is that there are no docstrings for the methods in the relevant Python file. There are two options that come to mind:
* Continue to use the gen-files plugin to generate documentation pages automatically, but exclude Python files without docstrings or ones that are not significant (such as zstd, which you mentioned).
* Remove the gen-files plugin and its automatic generation script and manually add only the most important and central modules/classes for the pyiceberg project.


Regarding the problem of the menu expanding, I have also noticed this issue. It appears to be caused by the gen-files plugin and the way in which I've configured it. I will investigate further, taking into account the decision we made in the previous point.

Let me know what do you think :)

**Comment by Fokko** [comment]
> Continue to use the gen-files plugin to generate documentation pages automatically, but exclude Python files without docstrings or ones that are not significant (such as zstd, which you mentioned).

has my strong preference, doing it manually is not a scalable solution :)

**Pull request merged**